tap_linkedin_ads/client.py

```diff
@@ -160,17 +159,29 @@ def __enter__(self):
     def __exit__(self, exception_type, exception_value, traceback):
         self.__session.close()
 
+    # The following two functions are used solely by unittests and are not utilized by the tap
+
+    def get_expires_time(self):
+        return self.__expires
+
+    def set_mock_expires(self, mock_expire):
+        self.__expires = mock_expire
+        return self.__expires
+
+
     @backoff.on_exception(backoff.expo,
-                          Server5xxError,
+                          (Server5xxError, LinkedInUnauthorizedError),
                           max_tries=5,
                           factor=2)
     def fetch_and_set_access_token(self):
         """
-        This method generates new access token if the refresh token is provided.
-
+        This method:
```

Comment:
This function is really tricky to mentally parse imo, there are a lot of if statements that branch. I think ideally the function would match the docstring more closely or would be broken up into multiple functions. Maybe a "check_for_valid_access_token" function that returns true if we have a token and it is not out of date (false otherwise)?

Might not be worth the time for one tap, so i'll leave the choice up to you.